protobuf/lib/src/protobuf/coded_buffer_writer.dart

```diff
@@ -141,22 +141,32 @@ class CodedBufferWriter {
           }
           buffer[outPos++] = v;
         } else {
-          // action is an amount of bytes to copy from _outputChunks into the
-          // buffer.
+          // `action` is an amount of bytes to copy from `_outputChunks` into
+          // the buffer.
           var bytesToCopy = action;
           while (bytesToCopy > 0) {
             final Uint8List chunk = _outputChunks[chunkIndex];
             final int bytesInChunk = _outputChunks[chunkIndex + 1];
 
-            // Copy at most bytesToCopy bytes from the current chunk.
+            // Copy at most `bytesToCopy` bytes from the current chunk.
             final leftInChunk = bytesInChunk - chunkPos;
             final bytesToCopyFromChunk =
                 leftInChunk > bytesToCopy ? bytesToCopy : leftInChunk;
             final endPos = chunkPos + bytesToCopyFromChunk;
-            while (chunkPos < endPos) {
-              buffer[outPos++] = chunk[chunkPos++];
+
+            if (bytesToCopyFromChunk <= 20) {
```

Comment:
The "optimization" in the `if` branch is probably due to the `else` branch doing the view object allocation, is it?

Our `Uint8List.setRange()` API actually takes an optional `skipCount` parameter that says how many bytes to skip in the `from`. So why not do this:
```dart
buffer.setRange(outPos, outPos + bytesToCopyFromChunk, chunk, chunkPos);
chunkPos += bytesToCopyFromChunk;
outPos += bytesToCopyFromChunk;
bytesToCopy -= bytesToCopyFromChunk;
```
instead of the `if { manual loop } else { allocate view + setRange }`?

Reply:
Good idea, thanks. PTAL. I've also updated the benchmark results.